extern/grackle/CMakeLists.txt

```diff
@@ -35,11 +35,17 @@ set(grackle_SRCS
 )
 
 add_library(grackle ${grackle_SRCS})
-target_include_directories(grackle PRIVATE ${HDF5_INCLUDE_DIRS} ${MPI_C_INCLUDE_PATH})
+
+set(PRECISION_DEFINE GRACKLE_FLOAT_8)
+configure_file(${PROJECT_SOURCE_DIR}/config/grackle_float.in.h ${CMAKE_BINARY_DIR}/extern/grackle/grackle_deps/grackle_float.h)
+
 target_include_directories(grackle PUBLIC ${PROJECT_SOURCE_DIR}/grackle_repo/src/clib)
-target_include_directories(grackle PUBLIC ${CMAKE_BINARY_DIR}/extern/grackle)
+target_include_directories(grackle PUBLIC ${PROJECT_BINARY_DIR}/grackle_deps)
+target_include_directories(grackle PRIVATE ${HDF5_INCLUDE_DIRS} ${MPI_C_INCLUDE_PATH})
 target_link_libraries(grackle ${HDF5_LIBRARIES} ${MPI_C_LIBRARIES} gfortran)
-target_compile_definitions(grackle PRIVATE LINUX H5_USE_16_API CONFIG_BFLOAT_8 PIC)
+#Omit warning of deprecated functions (already from removed SPH-EXA)
+target_compile_definitions(grackle PUBLIC OMIT_LEGACY_INTERNAL_GRACKLE_FUNC)
```

Comment:
This compile definition should probably be `PRIVATE`

Reply:
Problem is that the warnings are generated in grackle.h which is included by SPH-EXA.